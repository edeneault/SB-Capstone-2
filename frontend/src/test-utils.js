import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
