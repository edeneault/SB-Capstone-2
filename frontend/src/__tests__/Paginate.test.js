import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";

import Paginate from "../components/Paginate";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  let pages, page, keyword;
  beforeEach(() => {
    pages = 2;
    page = 1;
    keyword = "";
  });

  test("renders Paginate component without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <Paginate pages={pages} page={page} keyword={keyword} />
      </Provider>,
    );

    expect(getByText("1")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <Paginate pages={pages} page={page} keyword={keyword} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
