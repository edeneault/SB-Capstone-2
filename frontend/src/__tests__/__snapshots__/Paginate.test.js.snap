// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="pagination"
  >
    <li
      class="page-item active"
    >
      <span
        class="page-link"
        href="/page/1"
      >
        1
        <span
          class="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="/page/2"
      >
        2
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
